### PR TITLE
Add: Ability to Automatically Add Gmail MX Records

### PR DESCRIPTION
**Pull request by martinmckenna**
## Description

I had this branch sitting for a long while - finally putting it up for review

This allows someone to create Gmail MX records along with their new domain. Asked for by a customer a few months back.

[See here for Google documentation](https://support.google.com/a/answer/140034?hl=en)

## Type of Change
- Non breaking change ('update', 'change')

## To Test

1. Open the _add domain_ drawer
2. Toggle the "create Gmail MX records" checkbox

## Notes

This also adds loading text next to the button, so the user has a better idea of what is loading at that particular time

### CODE REVIEW

File: packages/manager/src/features/Domains/DomainDrawer.tsx [hunk 1]
```diff
@@ -629,17 +795,28 @@ class DomainDrawer extends React.Component<CombinedProps, State> {
...
                 return this.redirectToLandingOrDetail(type, domainData.id, {
                   recordError:
-                    'There was an issue creating default domain records.'
+                    /**
+                     * if domainData.error exists, it means there was an issue creating
+                     * Gmail MX records
+                     */
+                    domainData.error
+                      ? 'There were issues creating some Gmail MX records and default domain records.'
+                      : 'There was an issue creating default domain records.'
```
Comment (Jskobos) [inline packages/manager/src/features/Domains/DomainDrawer.tsx]: John and I noticed that this error persists across refreshes. You should add `history.replace(location.pathname, {});` to cDM or useEffect in DomainsDetail to prevent this. See line 153 of SupportTicketDetail.tsx

File: packages/manager/src/components/Button/Button.tsx [hunk 2]
```diff
@@ -42,6 +43,9 @@ const styles = (theme: Theme) =>
       }
     },
     root: {
+      minWidth: '105px',
```
Comment (johnwcallahan) [inline packages/manager/src/components/Button/Button.tsx]: I left feedback on this here: https://github.com/linode/manager/pull/5183#discussion_r301109412

At the very least, this change should be its own PR so that this site-wide change is easily tracked.

Comment (martinmckenna) [reply packages/manager/src/components/Button/Button.tsx]: yep - this change was intentional to get away from the jarring width change when you go from a button with the loading SVG to the button with text

Comment (martinmckenna) [reply packages/manager/src/components/Button/Button.tsx]: I'll make another PR

File: packages/manager/src/features/Domains/DomainDrawer.tsx (hunk 1 above)
Comment (martinmckenna) [reply packages/manager/src/features/Domains/DomainDrawer.tsx]: Added it in App.tsx since adding it at a component level doesn't make much sense. That means it'll be clearing out the error when you are navigated to that page, regardless of whether it's during a refresh or not.

**Comment by martinmckenna** [comment]
@Jskobos addressed your first 3 points - for the rest of the stuff, I agree that it can be iterated upon and we can probably add it somewhere to the detail view as well

**Comment by martinmckenna** [comment]
@johnwcallahan ready again

File: packages/manager/src/App.tsx [hunk 3]
```diff
@@ -73,10 +73,17 @@ export class App extends React.Component<CombinedProps, State> {
   }
 
   componentDidMount() {
+    const { history, location } = this.props;
+
+    /* 
+     clear state each time the app mounts 
+     so that we're not persisting error messages across refreshes
+    */
+    history.replace(location.pathname, {});
```
Comment (johnwcallahan) [inline packages/manager/src/App.tsx]: This has unintended side effects. Namely, it replaces any query params on page load, so e.g. clicking on http://localhost:3000/linodes/create?type=One-Click will take you to the Create flow with nothing preselected. Similar for the search page landing, navigating through OBJ buckets, etc.

It also changes the behavior of a few components, like SupportLink. Currently when you click on a SupportLink (e.g. to cancel Managed) the ticket drawer will be opened with the ticket pre-filled. This currently survives page refreshes, which is nice.

IMO it should be up to individual components to clear error state if they're using `history.state` for error messages.

File: packages/manager/src/components/CheckBox/CheckBox.tsx [hunk 4]
```diff
@@ -104,6 +107,7 @@ const LinodeCheckBox: React.StatelessComponent<FinalProps> = props => {
           label={props.text}
         />
         {toolTipText && <HelpIcon text={toolTipText} />}
+        {subtext && <Typography>{subtext}</Typography>}
```
Comment (johnwcallahan) [inline packages/manager/src/components/CheckBox/CheckBox.tsx]: Minor: might be more semantic to make this another Label.

File: packages/manager/src/features/Domains/DomainDrawer.tsx [hunk 5]
```diff
@@ -606,14 +772,22 @@ class DomainDrawer extends React.Component<CombinedProps, State> {
...
         if (type === 'master') {
           if (defaultRecordsSetting === 'linode') {
+            this.setState({ loadingText: 'Creating some default records...' });
             return generateDefaultDomainRecords(
               domainData.domain,
               domainData.id,
-              path(['ipv4', 0], selectedDefaultLinode),
-              path(['ipv6'], selectedDefaultLinode)
+              /**
+               * we did undefined checking above so selectedDefaultLinode is
```
Comment (johnwcallahan) [inline packages/manager/src/features/Domains/DomainDrawer.tsx]: I still think it's a good idea to check for `selectedDefaultLinode`, maybe in the `if (defaultRecordsSettings === 'linode')` condition above.

The null checking above is within another conditional 100+ lines back, so IMO it's pretty brittle. I left a similar comment in @Jskobos PR here: https://github.com/linode/manager/pull/5797... I think we should strive to trust TypeScript here and not overuse the not-null assertion operator.

File: packages/manager/src/features/Domains/DomainRecordDrawer.tsx [hunk 6]
```diff
@@ -220,7 +211,11 @@ class DomainRecordDrawer extends React.Component<CombinedProps, State> {
   );
 
   TargetField = ({ label }: { label: string }) => (
-    <this.TextField field="target" label={label} />
+    <this.TextField
+      field="target"
+      label={label}
+      helperText="Trailing periods are optional"
```
Comment (johnwcallahan) [inline packages/manager/src/features/Domains/DomainRecordDrawer.tsx]: This component is used in several places; not sure we want this helper text everywhere:

Screen Shot 2019-11-21 at 9 36 12 AM
Screen Shot 2019-11-21 at 9 38 49 AM

**Comment by martinmckenna** [comment]
closing because I won't have the time to fix these issues
